ec-europa/eubfr-data-lake
# refactor(value-store-projects): replace dynamodb with elasticsearch - EUBFR-65

Summary of changes:

- Streams API approach preserved
- events and lib preserved, only client API (dynamo to es) has changed
- covers CRUD on value-store, could be tested with existing demo of `eubfr-${stage}-demo-budg-dashboard` app
- elastic search api endpoint is feeded automatically in serverless env by a split of service
- deployment of elasticsearch service takes around 10 minutes
- when es is deployed, it will be available to the [dashboard](https://eu-central-1.console.aws.amazon.com/es/home?region=eu-central-1#) where you can see the service endpoint

Testing scenario:

- increment `stage` in `config.json`
- `yarn deploy`
- `yarn deploy-demo`
- check all is deployed as before
- find the root API endpoint from ES dashboard
- Send a quick `GET` request to `${API}/_search` to see no records state
- use the dashboard app to upload a file for a given ETL
- Send another `GET` request to the search API to validate input of records
- Try to update the file in the existing flow in the dashboard app
- Delete the file from the dashboard app
- Send a request to the search API for the last time to see records are cleared

Additions as of 13th Nov 2017:

- I've added a [commit](https://github.com/ec-europa/eubfr-data-lake/pull/40/commits/91ba30a22a48052eff78c6590851768eab63b83a) regarding another related ticket EUBFR-66
- This mapping is added because in @yhuard's [comment](https://webgate.ec.europa.eu/CITnet/jira/browse/EUBFR-41?focusedCommentId=2381180&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-2381180) both [Geo Distance Aggregation](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-geodistance-aggregation.html) and [GeoHash grid Aggregation ](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-geohashgrid-aggregation.html) require mapping of the geo data to be `geo_point`

Thus, to test EUBFR-66 in this pull request, verify this mapping on the index after uploading data:

![amazon elasticsearch service management console](https://user-images.githubusercontent.com/1923476/32736342-7d169268-c897-11e7-9283-abc1f989c4e1.png)
